{"cells":[{"cell_type":"markdown","id":"22caaa44-c27e-4d29-b5ee-49dca4cc7cf3","metadata":{},"source":["<p style=\"text-align:center\">\n","    <a href=\"https://skills.network/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDS0321ENSkillsNetwork865-2022-01-01\" target=\"_blank\">\n","    <img src=\"https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/assets/logos/SN_web_lightmode.png\" width=\"200\" alt=\"Skills Network Logo\"  />\n","    </a>\n","</p>\n","\n","<h1 align=center><font size = 5>Assignment: SQL Notebook for Peer Assignment</font></h1>\n","\n","Estimated time needed: **60** minutes.\n","\n","## Introduction\n","Using this Python notebook you will:\n","\n","1.  Understand the Spacex DataSet\n","2.  Load the dataset  into the corresponding table in a Db2 database\n","3.  Execute SQL queries to answer assignment questions \n"]},{"cell_type":"markdown","id":"1ae8e773-a2d5-4721-b534-598518e7c7e7","metadata":{},"source":["## Overview of the DataSet\n","\n","SpaceX has gained worldwide attention for a series of historic milestones. \n","\n","It is the only private company ever to return a spacecraft from low-earth orbit, which it first accomplished in December 2010.\n","SpaceX advertises Falcon 9 rocket launches on its website with a cost of 62 million dollars wheras other providers cost upward of 165 million dollars each, much of the savings is because Space X can reuse the first stage. \n","\n","\n","Therefore if we can determine if the first stage will land, we can determine the cost of a launch. \n","\n","This information can be used if an alternate company wants to bid against SpaceX for a rocket launch.\n","\n","This dataset includes a record for each payload carried during a SpaceX mission into outer space.\n"]},{"cell_type":"markdown","id":"8a3770ec-da5d-4ee2-b913-12094cd3dfb0","metadata":{},"source":["### Download the datasets\n","\n","This assignment requires you to load the spacex dataset.\n","\n","In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. Click on the link below to download and save the dataset (.CSV file):\n","\n"," <a href=\"https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBM-DS0321EN-SkillsNetwork/labs/module_2/data/Spacex.csv\" target=\"_blank\">Spacex DataSet</a>\n","\n"]},{"cell_type":"markdown","id":"99833862-9852-410a-9211-b655a30013b0","metadata":{},"source":["**Navigate to the Go to UI screen** \n","\n","* Refer to this insruction in this <a href=\"https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/Labs_Coursera_V5/labs/Lab%20-%20Sign%20up%20for%20IBM%20Cloud%20-%20Create%20Db2%20service%20instance%20-%20Get%20started%20with%20the%20Db2%20console/instructional-labs.md.html\">link</a> for viewing  the   Go to UI screen. \n","\n","\n","* Later click on **Data link(below SQL)**  in the Go to UI screen  and click on **Load Data** tab.  \n","\n","\n","\n","* Later browse for the downloaded spacex file.\n","\n","\n","\n","<img src=\"https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBM-DS0321EN-SkillsNetwork/labs/module_2/images/browsefile.png\" width=\"800\">\n","\n","\n","* Once done select the schema andload the file.  \n","\n","\n"," <img src=\"https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBM-DS0321EN-SkillsNetwork/labs/module_2/images/spacexload3.png\" width=\"800\">\n"," \n"]},{"attachments":{},"cell_type":"markdown","id":"f76e14f8-fa6a-445f-b2cc-7a4cf400a353","metadata":{},"source":["\n","If you are facing a problem in uploading the dataset (which is a csv file), you can follow the steps below to upload the .sql file instead of the CSV file:\n","\n","* Download the file <a href=\"https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBM-DS0321EN-SkillsNetwork/datasets/Spacex%20.sql\">Spacex.sql</a>\n","\n","* Later click on **SQL** in the  **Go to UI Screen**.\n","\n","* Use the **From file** option to browse for the **SQL** file and upload it.\n","\n","<img src=\"https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBM-DS0321EN-SkillsNetwork/labs/module_2/images/sqlfile.png\">\n","\n","* Once you upload the script,you can use the **Run All** option to run all the queries to insert the data.\n","\n","<img src=\"https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBM-DS0321EN-SkillsNetwork/labs/module_2/images/runall.png\">\n","\n","    \n"]},{"cell_type":"code","execution_count":8,"id":"80dbd24e-b3c4-456b-a0a0-e6b252d5172f","metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["Requirement already satisfied: sqlalchemy==1.3.9 in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (1.3.9)\n","Collecting ibm_db_sa\n","  Using cached ibm_db_sa-0.3.9-py3-none-any.whl (30 kB)\n","Collecting ibm-db>=2.0.0\n","  Using cached ibm_db-3.1.4.tar.gz (1.4 MB)\n","  Installing build dependencies: started\n","  Installing build dependencies: finished with status 'done'\n","  Getting requirements to build wheel: started\n","  Getting requirements to build wheel: finished with status 'error'\n"]},{"name":"stderr","output_type":"stream","text":["  error: subprocess-exited-with-error\n","  \n","  × Getting requirements to build wheel did not run successfully.\n","  │ exit code: 1\n","  ╰─> [27 lines of output]\n","      Detected 64-bit Python\n","      Traceback (most recent call last):\n","        File \"C:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py\", line 351, in <module>\n","          main()\n","        File \"C:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py\", line 333, in main\n","          json_out['return_val'] = hook(**hook_input['kwargs'])\n","        File \"C:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py\", line 118, in get_requires_for_build_wheel\n","          return hook(config_settings)\n","        File \"C:\\Users\\pc\\AppData\\Local\\Temp\\pip-build-env-f_tfmltk\\overlay\\Lib\\site-packages\\setuptools\\build_meta.py\", line 338, in get_requires_for_build_wheel\n","          return self._get_build_requires(config_settings, requirements=['wheel'])\n","        File \"C:\\Users\\pc\\AppData\\Local\\Temp\\pip-build-env-f_tfmltk\\overlay\\Lib\\site-packages\\setuptools\\build_meta.py\", line 320, in _get_build_requires\n","          self.run_setup()\n","        File \"C:\\Users\\pc\\AppData\\Local\\Temp\\pip-build-env-f_tfmltk\\overlay\\Lib\\site-packages\\setuptools\\build_meta.py\", line 335, in run_setup\n","          exec(code, locals())\n","        File \"<string>\", line 125, in <module>\n","        File \"<string>\", line 87, in _getinstalledDb2Path\n","        File \"C:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\subprocess.py\", line 626, in getstatusoutput\n","          data = check_output(cmd, shell=True, text=True, stderr=STDOUT)\n","        File \"C:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\subprocess.py\", line 421, in check_output\n","          return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,\n","        File \"C:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\subprocess.py\", line 505, in run\n","          stdout, stderr = process.communicate(input, timeout=timeout)\n","        File \"C:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\subprocess.py\", line 1141, in communicate\n","          stdout = self.stdout.read()\n","        File \"C:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\codecs.py\", line 322, in decode\n","          (result, consumed) = self._buffer_decode(data, self.errors, final)\n","      UnicodeDecodeError: 'utf-8' codec can't decode byte 0x82 in position 84: invalid start byte\n","      [end of output]\n","  \n","  note: This error originates from a subprocess, and is likely not a problem with pip.\n","error: subprocess-exited-with-error\n","\n","× Getting requirements to build wheel did not run successfully.\n","│ exit code: 1\n","╰─> See above for output.\n","\n","note: This error originates from a subprocess, and is likely not a problem with pip.\n"]},{"name":"stdout","output_type":"stream","text":["Requirement already satisfied: ipython-sql in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (0.5.0)\n","Requirement already satisfied: sqlparse in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from ipython-sql) (0.4.3)\n","Collecting sqlalchemy>=2.0\n","  Using cached SQLAlchemy-2.0.7-cp310-cp310-win_amd64.whl (2.0 MB)\n","Requirement already satisfied: prettytable in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from ipython-sql) (3.6.0)\n","Requirement already satisfied: ipython-genutils in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from ipython-sql) (0.2.0)\n","Requirement already satisfied: six in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from ipython-sql) (1.16.0)\n","Requirement already satisfied: ipython in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from ipython-sql) (8.8.0)\n","Requirement already satisfied: typing-extensions>=4.2.0 in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from sqlalchemy>=2.0->ipython-sql) (4.4.0)\n","Requirement already satisfied: greenlet!=0.4.17 in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from sqlalchemy>=2.0->ipython-sql) (2.0.2)\n","Requirement already satisfied: pygments>=2.4.0 in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from ipython->ipython-sql) (2.11.2)\n","Requirement already satisfied: prompt-toolkit<3.1.0,>=3.0.11 in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from ipython->ipython-sql) (3.0.36)\n","Requirement already satisfied: pickleshare in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from ipython->ipython-sql) (0.7.5)\n","Requirement already satisfied: stack-data in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from ipython->ipython-sql) (0.2.0)\n","Requirement already satisfied: matplotlib-inline in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from ipython->ipython-sql) (0.1.6)\n","Requirement already satisfied: decorator in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from ipython->ipython-sql) (5.1.1)\n","Requirement already satisfied: backcall in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from ipython->ipython-sql) (0.2.0)\n","Requirement already satisfied: colorama in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from ipython->ipython-sql) (0.4.6)\n","Requirement already satisfied: jedi>=0.16 in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from ipython->ipython-sql) (0.18.1)\n","Requirement already satisfied: traitlets>=5 in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from ipython->ipython-sql) (5.7.1)\n","Requirement already satisfied: wcwidth in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from prettytable->ipython-sql) (0.2.5)\n","Requirement already satisfied: parso<0.9.0,>=0.8.0 in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from jedi>=0.16->ipython->ipython-sql) (0.8.3)\n","Requirement already satisfied: executing in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from stack-data->ipython->ipython-sql) (0.8.3)\n","Requirement already satisfied: pure-eval in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from stack-data->ipython->ipython-sql) (0.2.2)\n","Requirement already satisfied: asttokens in c:\\users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages (from stack-data->ipython->ipython-sql) (2.0.5)\n","Installing collected packages: sqlalchemy\n","  Attempting uninstall: sqlalchemy\n","    Found existing installation: SQLAlchemy 1.3.9\n","    Uninstalling SQLAlchemy-1.3.9:\n","      Successfully uninstalled SQLAlchemy-1.3.9\n","Successfully installed sqlalchemy-2.0.7\n"]}],"source":["!pip install sqlalchemy==1.3.9\n","!pip install ibm_db_sa\n","from sqlalchemy import create_engine\n","!pip install ipython-sql"]},{"cell_type":"markdown","id":"4d078618-5f50-48d9-9b65-20da01d2228c","metadata":{},"source":["### Connect to the database\n","\n","Let us first load the SQL extension and establish a connection with the database\n"]},{"cell_type":"code","execution_count":9,"id":"dbecf5cd-c2d8-4b82-b84a-244c41e72af4","metadata":{},"outputs":[],"source":["#%load_ext sql"]},{"cell_type":"code","execution_count":10,"metadata":{},"outputs":[],"source":["import csv, sqlite3\n","\n","con = sqlite3.connect(\"my_data1.db\")\n","cur = con.cursor()"]},{"cell_type":"code","execution_count":11,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["Traceback (most recent call last):\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sql\\connection.py\", line 45, in __init__\n","    engine = sqlalchemy.create_engine(\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\engine\\__init__.py\", line 479, in create_engine\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\engine\\strategies.py\", line 61, in create\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\engine\\url.py\", line 172, in _get_entrypoint\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\util\\langhelpers.py\", line 239, in load\n","sqlalchemy.exc.NoSuchModuleError: Can't load plugin: sqlalchemy.dialects:sqlite\n","\n","Connection info needed in SQLAlchemy format, example:\n","               postgresql://username:password@hostname/dbname\n","               or an existing connection: dict_keys([])\n","Traceback (most recent call last):\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sql\\magic.py\", line 196, in execute\n","    conn = sql.connection.Connection.set(\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sql\\connection.py\", line 70, in set\n","    cls.current = existing or Connection(descriptor, connect_args, creator)\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sql\\connection.py\", line 45, in __init__\n","    engine = sqlalchemy.create_engine(\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\engine\\__init__.py\", line 479, in create_engine\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\engine\\strategies.py\", line 61, in create\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\engine\\url.py\", line 172, in _get_entrypoint\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\util\\langhelpers.py\", line 239, in load\n","sqlalchemy.exc.NoSuchModuleError: Can't load plugin: sqlalchemy.dialects:sqlite\n","\n","Connection info needed in SQLAlchemy format, example:\n","               postgresql://username:password@hostname/dbname\n","               or an existing connection: dict_keys([])\n"]}],"source":["%sql sqlite:///my_data1.db"]},{"cell_type":"code","execution_count":12,"metadata":{},"outputs":[{"data":{"text/plain":["101"]},"execution_count":12,"metadata":{},"output_type":"execute_result"}],"source":["\n","import pandas as pd\n","df = pd.read_csv(\"https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBM-DS0321EN-SkillsNetwork/labs/module_2/data/Spacex.csv\")\n","df.to_sql(\"SPACEXTBL\", con, if_exists='replace', index=False,method=\"multi\")"]},{"cell_type":"markdown","id":"b6599c7d-66c8-4569-ac9e-9f57523a8cc1","metadata":{},"source":["\n","\n","\n","**DB2 magic in case of  UI service credentials.**\n","\n","\n","\n","<img src=\"https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBM-DS0321EN-SkillsNetwork/labs/module_2/images/servicecredentials.png\" width=\"600\">  \n","\n","* Use the following format.\n","\n","* Add security=SSL at the end\n","\n","**%sql ibm_db_sa://my-username:my-password@my-hostname:my-port/my-db-name?security=SSL**\n"]},{"cell_type":"code","execution_count":16,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["Traceback (most recent call last):\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sql\\connection.py\", line 45, in __init__\n","    engine = sqlalchemy.create_engine(\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\engine\\__init__.py\", line 479, in create_engine\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\engine\\strategies.py\", line 54, in create\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\engine\\url.py\", line 229, in make_url\n","    if v is None:\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\engine\\url.py\", line 288, in _parse_rfc1738_args\n","    for key, value in dict_items\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\engine\\url.py\", line 71, in __init__\n","    :meth:`_engine.URL.set` and\n","ValueError: invalid literal for int() with base 10: ''\n","\n","Connection info needed in SQLAlchemy format, example:\n","               postgresql://username:password@hostname/dbname\n","               or an existing connection: dict_keys([])\n","Traceback (most recent call last):\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sql\\magic.py\", line 196, in execute\n","    conn = sql.connection.Connection.set(\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sql\\connection.py\", line 70, in set\n","    cls.current = existing or Connection(descriptor, connect_args, creator)\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sql\\connection.py\", line 45, in __init__\n","    engine = sqlalchemy.create_engine(\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\engine\\__init__.py\", line 479, in create_engine\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\engine\\strategies.py\", line 54, in create\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\engine\\url.py\", line 229, in make_url\n","    if v is None:\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\engine\\url.py\", line 288, in _parse_rfc1738_args\n","    for key, value in dict_items\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sqlalchemy\\engine\\url.py\", line 71, in __init__\n","    :meth:`_engine.URL.set` and\n","ValueError: invalid literal for int() with base 10: ''\n","\n","Connection info needed in SQLAlchemy format, example:\n","               postgresql://username:password@hostname/dbname\n","               or an existing connection: dict_keys([])\n"]},{"ename":"NameError","evalue":"name 'bludb' is not defined","output_type":"error","traceback":["\u001b[1;31m---------------------------------------------------------------------------\u001b[0m","\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)","Cell \u001b[1;32mIn[16], line 2\u001b[0m\n\u001b[0;32m      1\u001b[0m get_ipython()\u001b[39m.\u001b[39mrun_line_magic(\u001b[39m'\u001b[39m\u001b[39msql\u001b[39m\u001b[39m'\u001b[39m, \u001b[39m'\u001b[39m\u001b[39mibm_db_sa://55fbc997-9266-4331-afd3-888b05e734c0.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:\u001b[39m\u001b[39m'\u001b[39m)\n\u001b[1;32m----> 2\u001b[0m \u001b[39m31929\u001b[39m\u001b[39m/\u001b[39mbludb\n","\u001b[1;31mNameError\u001b[0m: name 'bludb' is not defined"]}],"source":["%sql ibm_db_sa://55fbc997-9266-4331-afd3-888b05e734c0.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:\n","31929/bludb?security=SSL"]},{"cell_type":"markdown","id":"1e99228b-31e0-45ab-af44-5d0a98e2edd5","metadata":{},"source":["## Tasks\n","\n","Now write and execute SQL queries to solve the assignment tasks.\n","\n","### Task 1\n","\n","\n","\n","\n","##### Display the names of the unique launch sites  in the space mission\n"]},{"cell_type":"code","execution_count":14,"id":"36cb2cda-f45b-432a-9f96-333a7d577c75","metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["Traceback (most recent call last):\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sql\\magic.py\", line 196, in execute\n","    conn = sql.connection.Connection.set(\n","  File \"c:\\Users\\pc\\anaconda3\\envs\\myenv\\lib\\site-packages\\sql\\connection.py\", line 82, in set\n","    raise ConnectionError(\n","sql.connection.ConnectionError: Environment variable $DATABASE_URL not set, and no connect string given.\n","\n","Connection info needed in SQLAlchemy format, example:\n","               postgresql://username:password@hostname/dbname\n","               or an existing connection: dict_keys([])\n"]}],"source":["%sql select distinct(LAUNCH_SITE) from SPACEXTBL;"]},{"cell_type":"markdown","id":"0bade2d8-a5cf-4325-8419-8bcd03f4efd1","metadata":{},"source":["\n","### Task 2\n","\n","\n","#####  Display 5 records where launch sites begin with the string 'CCA' \n"]},{"cell_type":"code","execution_count":3,"id":"5f4740b9-7e45-495a-8aec-108d52124302","metadata":{},"outputs":[{"data":{"text/plain":["'c:\\\\Users\\\\pc\\\\Desktop\\\\IBM labs\\\\Capstone project\\\\EDA'"]},"execution_count":3,"metadata":{},"output_type":"execute_result"}],"source":["cwd"]},{"cell_type":"markdown","id":"5f3f3a95-f842-453f-9aec-770976af841d","metadata":{},"source":["### Task 3\n","\n","\n","\n","\n","##### Display the total payload mass carried by boosters launched by NASA (CRS)\n"]},{"cell_type":"code","execution_count":4,"id":"701d7054-5983-4dad-93a1-8738139c7980","metadata":{},"outputs":[{"ename":"NameError","evalue":"name 'pd' is not defined","output_type":"error","traceback":["\u001b[1;31m---------------------------------------------------------------------------\u001b[0m","\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)","Cell \u001b[1;32mIn[4], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m Spacex_table \u001b[39m=\u001b[39m pd\u001b[39m.\u001b[39mread_csv(\u001b[39m'\u001b[39m\u001b[39mSpacex - Spacex.csv\u001b[39m\u001b[39m'\u001b[39m)\n\u001b[0;32m      2\u001b[0m Spacex_table\n","\u001b[1;31mNameError\u001b[0m: name 'pd' is not defined"]}],"source":["Spacex_table = pd.read_csv('Spacex - Spacex.csv')\n","Spacex_table"]},{"cell_type":"markdown","id":"0e14f297-4165-4d28-b82b-c4e502bc7ef3","metadata":{},"source":["### Task 4\n","\n","\n","\n","\n","##### Display average payload mass carried by booster version F9 v1.1\n"]},{"cell_type":"code","execution_count":null,"id":"5c9846c3-e5dc-4e72-b507-aa2592a054fa","metadata":{},"outputs":[],"source":[]},{"cell_type":"markdown","id":"a45072a9-7730-4b69-97ed-d0268811c564","metadata":{},"source":["### Task 5\n","\n","##### List the date when the first successful landing outcome in ground pad was acheived.\n","\n","\n","_Hint:Use min function_ \n"]},{"cell_type":"code","execution_count":null,"id":"17e67b40-6d7d-4558-b5e9-662fe281a6b6","metadata":{},"outputs":[],"source":[]},{"cell_type":"markdown","id":"f4a27811-c954-4f53-a1e3-8590c09b761e","metadata":{},"source":["### Task 6\n","\n","##### List the names of the boosters which have success in drone ship and have payload mass greater than 4000 but less than 6000\n"]},{"cell_type":"code","execution_count":null,"id":"ffd9b828-f248-4857-98cb-0ba01dd3d4c3","metadata":{},"outputs":[],"source":[]},{"cell_type":"markdown","id":"64a3b1b5-efb7-4156-9918-28655779e419","metadata":{},"source":["### Task 7\n","\n","\n","\n","\n","##### List the total number of successful and failure mission outcomes\n"]},{"cell_type":"code","execution_count":null,"id":"6f541ead-8838-4db7-8f01-01577b69a76c","metadata":{},"outputs":[],"source":[]},{"cell_type":"markdown","id":"3ae7619c-a8c7-409d-b85d-26ce81236e36","metadata":{},"source":["### Task 8\n","\n","\n","\n","##### List the   names of the booster_versions which have carried the maximum payload mass. Use a subquery\n"]},{"cell_type":"code","execution_count":null,"id":"6d3ca7c3-923c-4609-804e-5a1450e19247","metadata":{},"outputs":[],"source":[]},{"cell_type":"markdown","id":"ed179e86-f977-4328-aad5-7be049fe8000","metadata":{},"source":["### Task 9\n","\n","\n","##### List the failed landing_outcomes in drone ship, their booster versions, and launch site names for in year 2015\n"]},{"cell_type":"code","execution_count":null,"id":"cedcbed9-4e5a-49fb-a18b-1c0d65696894","metadata":{},"outputs":[],"source":[]},{"cell_type":"markdown","id":"5f78519d-8725-411f-b42e-9d420d1c8b22","metadata":{},"source":["### Task 10\n","\n","##### Rank the count of landing outcomes (such as Failure (drone ship) or Success (ground pad)) between the date 2010-06-04 and 2017-03-20, in descending order\n"]},{"cell_type":"code","execution_count":null,"id":"12d7b73d-26de-48c5-befd-0f24b12e1b91","metadata":{},"outputs":[],"source":[]},{"cell_type":"markdown","id":"e8d71a31-f2cc-49ba-99cd-1856c388b093","metadata":{},"source":["### Reference Links\n","\n","* <a href =\"https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/Labs_Coursera_V5/labs/Lab%20-%20String%20Patterns%20-%20Sorting%20-%20Grouping/instructional-labs.md.html?origin=www.coursera.org\">Hands-on Lab : String Patterns, Sorting and Grouping</a>  \n","\n","*  <a  href=\"https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/Labs_Coursera_V5/labs/Lab%20-%20Built-in%20functions%20/Hands-on_Lab__Built-in_Functions.md.html?origin=www.coursera.org\">Hands-on Lab: Built-in functions</a>\n","\n","*  <a  href=\"https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/Labs_Coursera_V5/labs/Lab%20-%20Sub-queries%20and%20Nested%20SELECTs%20/instructional-labs.md.html?origin=www.coursera.org\">Hands-on Lab : Sub-queries and Nested SELECT Statements</a>\n","\n","*   <a href=\"https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/Module%205/DB0201EN-Week3-1-3-SQLmagic.ipynb\">Hands-on Tutorial: Accessing Databases with SQL magic</a>\n","\n","*  <a href= \"https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/Module%205/DB0201EN-Week3-1-4-Analyzing.ipynb\">Hands-on Lab: Analyzing a real World Data Set</a>\n","\n","\n"]},{"cell_type":"markdown","id":"fa1521fd-7c86-4eff-8f64-b51e5131fbf8","metadata":{},"source":["## Author(s)\n","\n","<h4> Lakshmi Holla </h4>\n"]},{"cell_type":"markdown","id":"572ec773-b1e0-413e-9397-c67c5894fc74","metadata":{},"source":["## Other Contributors\n","\n","<h4> Rav Ahuja </h4>\n"]},{"cell_type":"markdown","id":"7f4005d1-e800-4f7b-b002-9a578c750c15","metadata":{},"source":["## Change log\n","| Date | Version | Changed by | Change Description |\n","|------|--------|--------|---------|\n","| 2021-10-12 | 0.4 |Lakshmi Holla | Changed markdown|\n","| 2021-08-24 | 0.3 |Lakshmi Holla | Added library update|\n","| 2021-07-09 | 0.2 |Lakshmi Holla | Changes made in magic sql|\n","| 2021-05-20 | 0.1 |Lakshmi Holla | Created Initial Version |\n"]},{"cell_type":"markdown","id":"6a2bdd2d-d066-4ad5-9313-5e53830cd9f8","metadata":{},"source":["## <h3 align=\"center\"> © IBM Corporation 2021. All rights reserved. <h3/>\n"]}],"metadata":{"kernelspec":{"display_name":"myenv","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.10.9"}},"nbformat":4,"nbformat_minor":4}
